### PR TITLE
Prevent block place above 320

### LINKED ISSUE #867: Exception when placing blocks above the height limit
Placing a block above the dimension height limit causes an exception:
```
java.lang.ArrayIndexOutOfBoundsException: Index 24 out of bounds for length 24
	at java.base/java.util.ImmutableCollections$ListN.get(ImmutableCollections.java:680)
	at net.minestom.server.instance.DynamicChunk.getSection(DynamicChunk.java:109)
	at net.minestom.server.instance.Chunk.getSectionAt(Chunk.java:93)
	at net.minestom.server.instance.DynamicChunk.setBlock(DynamicChunk.java:71)
	at net.minestom.server.instance.InstanceContainer.UNSAFE_setBlock(InstanceContainer.java:131)
	at net.minestom.server.instance.InstanceContainer.placeBlock(InstanceContainer.java:166)
	at net.minestom.server.listener.BlockPlacementListener.listener(BlockPlacementListener.java:133)
	at net.minestom.server.listener.manager.PacketListenerManager.processClientPacket(PacketListenerManager.java:88)
```

This should be easy to fix by checking if the placed block is within the world limits inside of `BlockPlacementListener`.

### PR DESCRIPTION
**Pull request by Nuckerr**
This checks if the position of the block is greater or equal to 320 and then prevents it's placement if it does

Closes #867

### CODE REVIEW

File: src/main/java/net/minestom/server/listener/BlockPlacementListener.java [hunk 1]
```diff
@@ -131,6 +131,7 @@ public static void listener(ClientPlayerBlockPlacementPacket packet, Player play
         // BlockPlacementRule check
         Block resultBlock = playerBlockPlaceEvent.getBlock();
         final BlockPlacementRule blockPlacementRule = BLOCK_MANAGER.getBlockPlacementRule(resultBlock);
+        if(placementPosition.y() >= 320) return;
```
Comment (TogAr2) [inline src/main/java/net/minestom/server/listener/BlockPlacementListener.java:134]: You should probably not hardcode the number, use the dimension to get the maximum y value. Also you should probably move this to just after the initialization of `placementPosition` for a fast exit

File: src/main/java/net/minestom/server/listener/BlockPlacementListener.java [hunk 2]
```diff
@@ -28,6 +28,7 @@
 
 public class BlockPlacementListener {
     private static final BlockManager BLOCK_MANAGER = MinecraftServer.getBlockManager();
+    private static final double BLOCK_HEIGHT = 320.0;
```
Comment (TogAr2) [inline src/main/java/net/minestom/server/listener/BlockPlacementListener.java:31]: This isn't constant for every instance, use `instance.getDimensionType().getMaxY()`

File: src/main/java/net/minestom/server/listener/BlockPlacementListener.java [hunk 3]
```diff
@@ -128,6 +129,8 @@ public static void listener(ClientPlayerBlockPlacementPacket packet, Player play
             return;
         }
 
+        if(placementPosition.y() >= BLOCK_HEIGHT) return;
```
Comment (TogAr2) [inline src/main/java/net/minestom/server/listener/BlockPlacementListener.java:132]: I meant moving it to line 92, in which case the events would not be called, but in my opinion they shouldn't be in this case

**Comment by TheMode** [comment]
Could you add a test?

**Comment by Nuckerr** [comment]
Which package/file would you reccomend?

**Comment by TheMode** [comment]
Both the instance & entity package would work

**Comment by Nuckerr** [comment]
The unit test fails because set block also triggers this error. Where should I add this check

**Comment by Nuckerr** [comment]
bump

**Comment by mworzala** [comment]
Merged in with minestom-ce in #1937

**Pull request closed without merging**